tests/test_special_ops.py

```diff
@@ -374,3 +374,15 @@ def test_pad(shape, dtype, pad_mode, contiguous):
         res_out = torch.nn.functional.pad(ref_x, pad_params, pad_mode, pad_value)
 
     gems_assert_equal(ref_out, res_out)
+
+
+@pytest.mark.parametrize("start", [0, 1, 3])
+@pytest.mark.parametrize("step", [1, 2, 5])
+@pytest.mark.parametrize("end", [128, 256, 1024])
+@pytest.mark.parametrize("dtype", [torch.int64, torch.float32])
```

Comment:
dtype could include all float types, and I suggest testing int types and None as well